src/coreComponents/mesh/ElementType.hpp

```diff
@@ -40,8 +41,41 @@ enum class ElementType : integer
   Polyhedron     ///< General polyhedral element
 };
 
+/**
+ * @brief @return number of supported element types
+ * @note this MUST be updated if a new element type is inserted after Polyhedron
+ */
+inline constexpr integer numElementTypes()
```

Comment:
Ok. I have to admit, I didn't want to add this function at all. Ideally the number of enums could be extracted from `EnumStrings`, which has this information (assuming the string mapping is correct). But current implementation is not `constexpr` (because of local `static`), which means I can't use it as an `std::array<>` bound, even though this is compile-time knowledge. Yet another thing to try to improve.

Reply:
The problem with the `ENUM_STRINGS` is that we can improve it, but it may not be perfectly standard if we want a perfect macro... Don't know what's best...